develop: move ForLogrus to separate file and support AnyTags

vintorez opened the pull request.

Commit:
develop: move ForLogrus to separate file and support AnyTags

xaionaro reviewed (changes requested):
It would be better (faster) to use `Each` instead of `ToFastTags`

Commit:
fix: use Each instead of ToFastTags

vintorez commented:
done

xaionaro commented on helpers.go:
```diff
@@ -0,0 +1,12 @@
+
+import "github.com/sirupsen/logrus"
+
+func AnyTagsToLogrusFields(tags AnyTags) logrus.Fields {
```
I think this's a too long name for a helper. Imagine how it will look in the code.

However if you think it's fine then OK :)

vintorez replied:
yes, I agree with you, I'll fix it. and I found another issue with this helper, don't merge it!

xaionaro commented:
@Vintorez also please check if `tags == nil`

Commit:
develop: reformat TagsToMap function

Commit:
fix: check if tags == nil

vintorez commented:
removed all dependencies and changed TagsToMap function to return map[string]interface{}. it is a base type for all fields types like logrus.Fields and echolog.JSON

Commit:
develop: set ToMap() as a method of AnyTags interface

xaionaro commented on fast_tags.go:
```diff
@@ -160,3 +160,19 @@ func (tags FastTags) Each(fn func(k string, v interface{}) bool) {
+		}
+	}
+	if tags != nil {
+		tags.Each(func(k string, v interface{}) bool {
```
fieldMaps has priority of tags. In your case it's conversely.

vintorez replied:
fixed

xaionaro commented on tags.go:
```diff
@@ -186,3 +173,19 @@ func (tags Tags) ToFastTags() *FastTags {
+		}
+	}
+	if tags != nil {
+		tags.Each(func(k string, v interface{}) bool {
```
1. The same problem with priorities.
2. You use `Each` while you know that it's just a map here. Why?
3. You completely repeat the code while it could be implemented as a common function for both methods.

Fixes for p.2 and p.3 are mutually exclusive, choose :)

vintorez replied:
1. fixed
2&3. I chose the second option )

Commit:
develop: change ToMap() implementations
